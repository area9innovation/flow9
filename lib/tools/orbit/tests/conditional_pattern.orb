// Test conditional pattern matching with orbit2sexpr conversion

// Define a function that uses conditional pattern matching
fn classify(n) = 
  n is (
    x => "greater than 20"  if  x > 20;
    x => "between 11 and 20"  if  x > 10;
    x => "less than 5"  if  x < 5;
    x => "between 5 and 10"
  );

// Test with different values
println("5: " + classify(5));
println("3: " + classify(3));
println("15: " + classify(15));
println("25: " + classify(25));