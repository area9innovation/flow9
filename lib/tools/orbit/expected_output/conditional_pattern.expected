5: between 5 and 10
3: less than 5
15: between 11 and 20
25: greater than 20
Result: 0
